pkg/configuration/config.go

```diff
@@ -59,7 +59,12 @@ func init() {
 }
 
 // Setup the config directory and config file
-func Setup() {
+func Setup(cmd *cobra.Command) {
+	if cmd.Flags().Changed("config-dir") {
+		UserConfigFile = filepath.Join(UserConfigDir, configFileName)
+	}
```

Comment:
Is this change needed if you're setting the `UserConfigDir` and `UserConfigFile` in root's `loadFlags()` function?

Reply:
I'm not familiar enough with Cobra to know if it _should_ be required, but I know in practice it seems to be. With that change backed out:

```
❯ ./doppler secrets -p example -c dev --config-dir /tmp/doppler --debug
Debug: Using config dir /tmp/doppler
Debug: Using config file /Users/watsonian/.doppler/.doppler.yaml
```

with the change in place:

```
❯ ./doppler secrets -p example -c dev --config-dir /tmp/doppler --debug
Debug: Using config dir /tmp/doppler
Debug: Using config file /tmp/doppler/.doppler.yaml
```

Reply:
I pushed up a [change](https://github.com/DopplerHQ/cli/compare/9ba954ce6859fdcc34e64a7a0060047273a7e403..4ba5985d517e2d11dffd3561e089798b2d12d035) to simplify this a little bit. I also added some tests to verify that `--config-dir` and `--configuration` play nicely.